src/main/java/io/confluent/idesidecar/restapi/resources/TemplateResource.java

```diff
@@ -29,30 +46,65 @@
 @Path(TemplateResource.API_RESOURCE_PATH)
 @Produces(MediaType.APPLICATION_JSON)
 @Tag(name = "Templates", description = "Code generation templates")
+@Blocking
+@ApplicationScoped
 public class TemplateResource {
 
   public static final String API_RESOURCE_PATH = "/gateway/v1/templates";
 
-  @Inject
-  TemplateRegistryService templateRegistryService;
+  static final String SCAFFOLDING_SERVICE_LIST_TEMPLATES_URI = ConfigProvider
+      .getConfig()
+      .getValue("ide-sidecar.scaffolding-service.list-templates-uri", String.class);
+  static final String SCAFFOLDING_SERVICE_GET_TEMPLATE_URI = ConfigProvider
+      .getConfig()
+      .getValue("ide-sidecar.scaffolding-service.get-template-uri", String.class);
+  static final String SCAFFOLDING_SERVICE_APPLY_TEMPLATE_URI = ConfigProvider
+      .getConfig()
+      .getValue("ide-sidecar.scaffolding-service.apply-template-uri", String.class);
+
+  static final MultiMap NO_HEADERS = MultiMap.caseInsensitiveMultiMap();
+  static final Map<String, String> NO_PATH_PARAMS = Map.of();
+
+  static final ProxyContext LIST_TEMPLATES_PROXY_CONTEXT = new ProxyContext(
+      SCAFFOLDING_SERVICE_LIST_TEMPLATES_URI,
+      NO_HEADERS,
+      HttpMethod.GET,
+      null,
+      NO_PATH_PARAMS,
+      // No connection ID
+      null
+  );
+
+  static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
 
   @Inject
-  TemplateEngine templateEngine;
+  @Named("scaffoldingProxyProcessor")
+  Processor<ProxyContext, Future<ProxyContext>> scaffoldingProxyProcessor;
 
   @GET
-  public TemplateList list() {
-    var templateManifests = templateRegistryService.listTemplates();
-    var templateList = templateManifests.stream()
-        .map(Template::new)
-        .sorted(Comparator.comparing(Template::id))
-        .toList();
-    return new TemplateList(templateList);
+  public Uni<TemplateList> list() {
+    return uniStage(() -> scaffoldingProxyProcessor.process(LIST_TEMPLATES_PROXY_CONTEXT).toCompletionStage())
+        .chain(context -> parseJson(
+            context.getProxyResponseBody().toString(), ScaffoldV1TemplateList.class))
```

Comment:
Did you decide to parse the CCloud service's JSON response and return a `Uni<TemplateList>` here (and in `get()`) so that we can auto-generate the OpenAPI spec?

Reply:
Yeah, it's so that there's no migration needed from the VS Code calling side.